statsmodels/tsa/tests/test_tsa_indexes.py

```diff
@@ -710,6 +739,105 @@ def test_prediction_increment_pandas_dates_nanosecond():
     assert_equal(prediction_index.equals(desired_index), True)
 
 
+@pytest.mark.skipif(not has_range_index, reason='No RangeIndex')
+def test_range_index():
+    tsa_model.__warningregistry__ = {}
+
+    endog = pd.Series(np.random.normal(size=5))
+    assert_equal(isinstance(endog.index, pd.RangeIndex), True)
+    # Warning should not be given
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter('always')
+        mod = tsa_model.TimeSeriesModel(endog)
+        assert_equal(len(w), 0)
+
+
+@pytest.mark.skipif(not has_range_index, reason='No RangeIndex')
+def test_prediction_rangeindex():
+    index = supported_increment_indexes[2][0]
+    endog = pd.Series(dta[0], index=index)
+    mod = tsa_model.TimeSeriesModel(endog)
+
+    # Basic prediction: [0, end]
+    start_key = 0
+    end_key = None
+    start, end, out_of_sample, prediction_index = (
+        mod._get_prediction_index(start_key, end_key))
+
+    assert_equal(start, 0)
+    assert_equal(end, nobs - 1)
+    assert_equal(out_of_sample, 0)
+    desired_index = pd.RangeIndex(start=-5, stop=0, step=1)
+    assert_equal(prediction_index.equals(desired_index), True)
+
+    # Negative index: [-2, end]
+    start_key = -2
+    end_key = -1
+    start, end, out_of_sample, prediction_index = (
+        mod._get_prediction_index(start_key, end_key))
+
+    assert_equal(start, 3)
+    assert_equal(end, 4)
+    assert_equal(out_of_sample, 0)
+    desired_index = pd.RangeIndex(start=-2, stop=0, step=1)
+    assert_equal(prediction_index.equals(desired_index), True)
+
+    # Forecasting: [1, 5]
+    start_key = 1
+    end_key = nobs
+    start, end, out_of_sample, prediction_index = (
+        mod._get_prediction_index(start_key, end_key))
+
+    assert_equal(start, 1)
+    assert_equal(end, 4)
+    assert_equal(out_of_sample, 1)
+    desired_index = pd.RangeIndex(start=-4, stop=1, step=1)
+    assert_equal(prediction_index.equals(desired_index), True)
+
+
+@pytest.mark.skipif(not has_range_index, reason='No RangeIndex')
+def test_prediction_rangeindex_withstep():
```

Comment:
Same as above re asking for a comment.  Also this is a good use case for parametrizing these two tests and getting rid of a bunch of duplicated code.

Reply:
> Also this is a good use case for parametrizing these two tests and getting rid of a bunch of duplicated code.

I agree, but will leave it for later :)